kalite/distributed/static/js/distributed/inline/views.js

```diff
@@ -0,0 +1,95 @@
+window.ButtonView = Backbone.View.extend({
+    template: HB.template("inline/inline"),
+
+    initialize: function() {
+        _.bindAll(this);  
+        this.render();
+    },
+
+    events: {
+        "click": "clickCallback"
+    },
+
+    clickCallback: function() {
+        var self = this;
+        this.model.fetch({ 
+            success: function(model, response, options) {
+                //obtain narrative, JSON obj of elements and their attributes
+                var narr = self.model.attributes;
+
+                //translate narrative into build options for introjs
+                var options = self.parseNarrative(narr);
+
+                //set the options on introjs object
+                var intro = introJs();
+                intro.setOption('tooltipPosition', 'auto');
+                intro.setOption('positionPrecedence', ['left', 'right', 'bottom', 'top']);
+                intro.setOptions(options);
+
+                intro.start();
+            },
+
+            error: function(model, response, options) {
+                console.log("Unable to load inline tutorial narrative!");
+            }
+        });
+    },
+
+    //Translate narrative into JSON obj used to set introjs options
+    parseNarrative: function(narr) {
+        var options = {};
+        var steps = [];
+
+        var key = Object.keys(narr);
+        key = key[1];
+
+        // Parse narrative into obj using attribute keywords for intro framework
+        var newselectors = _.map(narr[key], function(element) {
+            var step = {};
+            var selectorKey = String(Object.keys(element));
+            var attributes = [];
+            attributes = element[selectorKey];
+
+            if (selectorKey != "unattached") {
+                step["element"] = selectorKey;
+            }
+
+            //Set {key: value} pairs that define modal for each intro step
+            var parsed = _.map(attributes, function(attribute) {
+                var key = Object.keys(attribute);
+                key = key[0];
+                var value = attribute[key];
+                var newkey = null;
+                var newvalue = null;
+
+                if (key === "text") {
+                    newkey = "intro";
+                    newvalue = value;
+                }
+                else if (key === "position") {
```

Comment:
I was looking at the [spec](https://docs.google.com/document/d/11wY1EViZrCKXZkus-7Pf4RMCbEYuaTNSCRa5YCglMMU/edit):

__Description of intro YAML format:__
Each url pattern has a list of jQuery-style selectors. Modals will be attached to each selector. Each selector has a list of attributes which determine the content and presentation of the modal. Valid attributes include:
* step: An integer. Should be sequential. Required.
* text: The text presented in the modal. Required.
* before-showing: Optional. User actions taken to make an element visible. A list of key-value pairs. Supposed to be extensible, in case we find we need more actions. Valid keys are:
  * click: Clicks an object specified by the value. Value is a jQuery-style selector.
  * scroll-to: An integer. Value is x-position on the page that should be scrolled to.

Which of the attributes we discussed corresponds to `position` here? We should also implement `before-showing`. I can't recall if we had another discussion about this and I failed to update the spec, if that's the case please update the spec.